### PR TITLE
Update Excel plugin with output support

### PR DESCRIPTION
**Pull request by JCotton1123**
Example profiles.yaml:

```yaml
     plugins:
        - module: excel
          alias: accounts_file
          config:
            output:
              engine: 'xlsxwriter'
              engine_kwargs:
                # See https://xlsxwriter.readthedocs.io/workbook.html#Workbook
                options:
                  # See https://xlsxwriter.readthedocs.io/working_with_memory.html#memory-perf
                  strings_to_numbers: true
                  remove_timezone: true
                  use_zip64: true
                  nan_inf_to_errors: true
              file: "files/accounts.xlsx"
              index: false  # Don't output row number
              header_styling: false  # Disable default Pandas header styling
```

Example model:

```
{{ config(materialized='external', plugin='accounts_file', overrides={'sheet_name': 'New Accounts'}) }}

SELECT *
FROM {{ ref('accounts') }}
WHERE status = 'NEW'
```

### CODE REVIEW

File: dbt/adapters/duckdb/plugins/excel.py [hunk 1]
```diff
@@ -29,3 +40,45 @@ def load(self, source_config: SourceConfig):
...
+
+
+    def store(self, target_config: TargetConfig):
+        plugin_output_config = self._config["output"]
+
+        # Create the writer on the first instance of the call to store.
+        # Instead if we instantiated the writer in the constructor
+        # with mode = 'w', this would result in an existing file getting
+        # overwritten. This can happen if dbt test is executed for example.
+        if not hasattr(self, '_excel_writer'):
```
Comment (jwills) [inline dbt/adapters/duckdb/plugins/excel.py:52]: Q: does this bit need a lock? I'm assuming it's possible for multiple threads to hit it at the same time.

**Comment by JCotton1123** [comment]
@jwills the one thing this PR is missing are tests. If you can provide some guidance there I am happy to implement some. Also, thanks again for taking the time to chat this morning.

**Comment by JCotton1123** [comment]
@jwills just getting back to this...I attempted to follow your suggestion for implementing a test. LMK if this needs any additional changes?

**Comment by JCotton1123** [comment]
@jwills I believe those issues should be resolved 🤞

**Comment by jwills** [comment]
@JCotton1123! Yay-- thanks again!

**Pull request merged**